File: server/src/main/com/broll/gainea/server/core/cards/DirectlyPlayedCard.java
package com.broll.gainea.server.core.cards;

import com.broll.gainea.net.NT_Card;

public abstract class DirectlyPlayedCard extends Card {
    public DirectlyPlayedCard(int picture, String title, String text) {
        super(picture, title, text);
    }

    @Override
    public boolean isPlayable() {
        return true;
    }

    @Override
    public NT_Card nt() {
        NT_Card nt = super.nt();
        nt.playable = false;
        return nt;
    }
}

File: server/src/main/com/broll/gainea/server/core/cards/DirectlyPlayedCard.kt
package com.broll.gainea.server.core.cards

import com.broll.gainea.net.NT_Card

abstract class DirectlyPlayedCard(
    picture: Int,
    effectType: EffectType,
    title: String,
    text: String
) : Card(picture, effectType, title, text) {
    override val isPlayable: Boolean
        get() = true

    override fun nt(): NT_Card {
        val nt = super.nt()
        nt.playable = false
        return nt
    }
}
